modules/core/src/v2/coins/abstractUtxoCoin.ts

```diff
@@ -1192,22 +1207,72 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
         debug('Input details: %O', signatureContext);
 
         const sigHashType = utxolib.bitgo.getDefaultSigHash(self.network);
-        const redeemScript = signatureContext.unspent.redeemScript
-          ? Buffer.from(signatureContext.unspent.redeemScript, 'hex')
-          : undefined;
-        const witnessScript = signatureContext.unspent.witnessScript
-          ? Buffer.from(signatureContext.unspent.witnessScript, 'hex')
-          : undefined;
-        const scriptType = witnessScript ? (redeemScript ? 'p2shP2wsh' : 'p2wsh') : 'p2sh';
-        debug(`Signing ${scriptType} input`);
-        try {
-          txb.sign(index, keyPair, redeemScript, sigHashType, signatureContext.unspent.value, witnessScript);
-        } catch (e) {
-          debug('Failed to sign input:', e);
-          signatureContext.error = e;
-          signatureIssues.push(signatureContext);
-          continue;
+        if (Codes.isP2tr(signatureContext.chain)) {
+          // we default to signing for a p2tr-p2ns script path spend using
+          // the bitgo key and user key
+
+          // we need to provide the sign call with the control block and chosen tapscript
+          if (!params.userKeychain || !params.backupKeychain || !params.bitgoKeychain) {
+            throw new Error('missing keychain parameters');
+          }
+
+          // we expect that the first signature will always be with the user key
+          const userPubkey = bip32
+            .fromBase58(params.userKeychain.pub)
+            .derivePath(sanitizeLegacyPath(signatureContext.path)).publicKey;
+          const backupPubkey = bip32
+            .fromBase58(params.backupKeychain.pub)
+            .derivePath(sanitizeLegacyPath(signatureContext.path)).publicKey;
+          const bitgoPubkey = bip32
+            .fromBase58(params.bitgoKeychain.pub)
+            .derivePath(sanitizeLegacyPath(signatureContext.path)).publicKey;
+
+          const { controlBlock, redeem, output } = utxolib.bitgo.outputScripts.createPaymentP2tr(
+            [userPubkey, backupPubkey, bitgoPubkey],
+            params.taprootRedeemIndex ?? 0
+          );
+          prevOutputs.push({
+            prevOutScript: output!,
+            value: signatureContext.unspent.value,
+          });
```

Comment:
we won't reach this for replay protection puts

let's use `tx.ins.map()` instead

Reply:
for now let's go with a simpler change though